### ISSUE TITLE
🐛 Signature for Trajectory.on_env_feedback() Interface should include the arguments used by the Actor class

### ISSUE THREAD

**Issue by jreynolds01** [microsoft/maro#373]
## Description

The Trajectory.on_env_feedback() interface defined [here](https://github.com/microsoft/maro/blob/a61968b9e53ab83e248ec258dd8ea7c05c56bf4c/maro/rl/training/trajectory.py#L21)
only includes self as an argument.

However, the Actor() class defined [here](https://github.com/microsoft/maro/blob/a61968b9e53ab83e248ec258dd8ea7c05c56bf4c/maro/rl/training/actor.py#L65) requires that the on_env_feedback() method include additional arguments:

```
self.trajectory.on_env_feedback(
                event, state_by_agent, action_by_agent, reward if reward is not None else self.env.metrics
            )
```

When we define a custom Trajectory, we can overwrite this method (and we do), but the difference in signature produces pylint errors:

```
W0221:Parameters differ from overridden 'on_env_feedback' method
```

As far as I can tell, every use of `on_env_feedback` requires that same signature, so it should be represented in the interface.

## To Reproduce

1. Write a custom
2.
3.

## Environment

- MARO version (e.g., v0.1.1a1):
- MARO scenario (`CIM`, `Citi Bike`):
- MARO component (`Simulation`, `RL`, `Distributed Training`):
- Orchestration platform (`GraSS on Azure`, `AKS on Azure`):
- How you installed MARO (`pip`, `source`):
- OS (`Linux`, `Windows`, `macOS`):
- Python version (`3.6`, `3.7`):
- Docker image (e.g., maro2020/maro:latest):
- CPU/GPU:
- Any other relevant information:

**Comment by ysqyang** [comment]
That is correct @jreynolds01. Thanks for pointing it out. FYI, Trajectory has been revamped to AbsEnvWrapper in the upcoming version, so things will appear quite different. Stay tuned.
